### PR TITLE
🎉  New Source Dockerhub

### PR DESCRIPTION
**Pull request by swyxio**
## What
*Describe what the change is solving*

adds new connector as requested in #12773

## How

as described in #12773

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

### Community member or Airbyter

- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. 
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*
image

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:47]: This is not expected to be here, right?

**Review by ChristopheDuong — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (sw-yx) [reply airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:47]: whoops sorry

File: airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml:27]:
```suggestion
```

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:41]: are we allowed/encouraged to print nice confirm messages like this?

File: airbyte-integrations/connectors/source-dockerhub/Dockerfile
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-dockerhub/Dockerfile:38]: How come you have so many leftover references to the source-python-http-example everywhere?

Is this automatically generated by the templated connectors? or did you copy paste from an example connector?

Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-dockerhub/Dockerfile:38]: How come you have so many leftover references to the source-python-http-example everywhere?

Is this automatically generated by the templated connectors? or did you copy paste from an example connector?

Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/Dockerfile:38]:
```suggestion
COPY source_dockerhub ./source_dockerhub

ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]

LABEL io.airbyte.version=0.1.0
LABEL io.airbyte.name=airbyte/source-dockerhub
```

File: airbyte-integrations/connectors/source-dockerhub/.dockerignore
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/.dockerignore:4]:
```suggestion
!source_dockerhub
```

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/spec.yaml
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/spec.yaml:1]:
```suggestion
documentationUrl: https://docs.airbyte.io/integrations/sources/dockerhub
```

File: airbyte-integrations/connectors/source-dockerhub/sample_files/config.json
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/sample_files/config.json:1]:
```suggestion
{
  "docker_username": "airbyte"
}
```

File: airbyte-integrations/connectors/source-dockerhub/sample_files/invalid_config.json
Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/sample_files/invalid_config.json:1]: {
  "invalid_username": "this should error because wrong name of field supplied"
}

Comment (sw-yx) [inline airbyte-integrations/connectors/source-dockerhub/sample_files/invalid_config.json:1]:
```suggestion
{
  "invalid_username": "this should error because wrong name of field supplied"
}
```

File: docs/integrations/sources/dockerhub.md
Comment (sw-yx) [inline docs/integrations/sources/dockerhub.md:35]:
```suggestion
```

**Review by ChristopheDuong — APPROVED** [review]
Maybe someone from the connector team could also give a look?

There will be few steps to go through to publish it before merging though

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (Phlair) [reply airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:41]: I'm actually surprised that this ^ is not in the docs!? We should definitely add it.

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:54]: nit: Since you're not actually passing any `**kwargs` in when you instantiate this you can just call super like `super().__init__()`

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:60]: this comment indicates that there is no pagination and we this function ought to just return None, however the code itself looks like it's handling pagination, is this comment just out of date?

File: airbyte-integrations/connectors/source-dockerhub/unit_tests/test_streams.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/unit_tests/test_streams.py:18]: Do we need this? `DockerHub` isn't an abstract class so it should already be implementing any abstract methods from `HttpStream`

File: airbyte-integrations/connectors/source-dockerhub/unit_tests/test_streams.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/unit_tests/test_streams.py:94]: nit: you could just remove these lines

**Review by Phlair — COMMENTED** [review]
Awesome! This is looking great!

I've left some minor comments around tidiness and standard practices but this is looking great.

The next steps will be to `/test` (and fix any bugs) and then `/publish` this connector! Let me know if you'd like to sync up on Monday and we could go through this flow together on a call 👍

File: airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml:8]: where these point to `secrets/config.json` should be replaced with `sample_files/config.json`.

Usually the config has to be in secrets due to it having secret values but that's not the case here so we can alter the pattern :)

File: airbyte-integrations/connectors/source-dockerhub/sample_files/invalid_config.json
Comment (sw-yx) [reply airbyte-integrations/connectors/source-dockerhub/sample_files/invalid_config.json:3]: i actually thought that that came from the generated template so i left them there untouched... understood

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (sw-yx) [reply airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:41]: i've put it in the #docs channel. 

i also think we should put this in the generated template so that people just get nudged to log properly. where can we do this?

**Comment by sw-yx** [comment]
looks like we now have a merge conflict in SUMMARY.md - anyone with write access can help resolve?

**Comment by Amruta-Ranade** [comment]
We don't need the SUMMARY.md file now that we've moved to Docusaurus.

**Review by Phlair — APPROVED** [review]
Looking great, thanks for making the changes 👍 

I pulled this down locally to get the acceptance tests working and in doing so noticed that each record was actually the full page response from the API. I think it makes more sense for each record to be each Docker image so [I made that change here](https://github.com/airbytehq/airbyte/pull/13007/commits/f0d5343d4196b02ce20014d834b29311de18e975). Let me know if you agree with this change or not :)

File: airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py:89]: @sw-yx this is the specific change on the records

**Comment by sw-yx** [comment]
hey @Phlair ! back on this again. i'm not replicating those transient issues - any ideas how to repro and/or do i just need to add some retries?

**Comment by Phlair** [comment]
Hey @sw-yx, I noticed it was to do with the "pull count" on docker images increasing between tests, so the test was recognising this as _different data_ and failing. Fixed that by [adding in ignore fields for the tests here](https://github.com/airbytehq/airbyte/pull/13007/commits/ef25420e7e208b29c94bc394a2ea965a01c7b9e8).

**Comment by sw-yx** [comment]
THANK YOU THANK YOU!!!

/publish connector=connectors/source-dockerhub

**Comment by sw-yx** [comment]
image
hmm the github action is failing because of some weird label. the error message is not very clear. any idea what this means @phlair?

**Comment by Phlair** [comment]
make sure to run `/publish connector=connectors/source-dockerhub` in a comment on its own with no other text 👍

**Comment by sw-yx** [comment]
@Phlair looks like its still failing, with the same error above

**Comment by Phlair** [comment]
@sw-yx I've moved this PR over [into the main repo here](https://github.com/airbytehq/airbyte/pull/13931). We should be able to publish it and have it run the CI checks properly then.
I'll close out this PR now and we'll move to the other one.

**Pull request closed without merging**
